tapgarden/caretaker.go

```diff
@@ -679,10 +679,32 @@ func (b *BatchCaretaker) stateStep(currentState BatchState) (BatchState, error)
 
 		b.cfg.Batch.RootAssetCommitment = tapCommitment
 
+		// Fetch the optional Tapscript tree for this batch, and convert
+		// it to a TapHash.
+		// TODO(jhb): Make this load call mandatory once we have a
+		// DB-backed implementation of the tapscript tree store.
+		var batchSibling *commitment.TapscriptPreimage
+		if b.cfg.TreeStore != nil && b.cfg.Batch.tapSiblingHash != nil {
+			tapSibling, err := b.cfg.TreeStore.LoadTapscriptTree(
+				ctx, *b.cfg.Batch.tapSiblingHash,
+			)
+			if err != nil {
+				return 0, err
+			}
+			if tapSibling == nil {
+				return 0, asset.TreeNotFound
```

Comment:
Should the underlying implementation return `ErrNoRows` instead?

Reply:
I thought that was only for queries expecting one result / tagged with `:one` ? I recall finding the docs for this before but haven't rediscovered them yet...

Reply:
Confirmed what I posted above in tests.